model/events/LtiAgsListener.php

```diff
@@ -108,60 +89,49 @@ public function onDeliveryExecutionResultsRecalculated(DeliveryExecutionResultsR
         }
     }
 
-    private function onDeliveryExecutionFinish(DeliveryExecutionState $event): void
+    public function onDeliveryExecutionFinish(TestVariablesRecorded $event): void
     {
-        /** @var User $user */
-        $user = $event->getContext()->getParameter(DeliveryExecutionStateContext::PARAM_USER);
-        $deliveryExecution = $event->getDeliveryExecution();
-
-        if ($user instanceof Lti1p3User) {
-            /** @var TestSessionService $testSessionService */
-            $testSessionService = $this->getServiceManager()->get(TestSessionService::SERVICE_ID);
-            $session = $testSessionService->getTestSession($deliveryExecution);
-
-            $scoreTotal = null;
-            $scoreTotalMax = null;
-
-            foreach ($session->getAllVariables()->getArrayCopy() as $variable) {
-                if ($variable instanceof OutcomeVariable) {
-                    $value = $variable->getValue();
-
-                    if (!$value instanceof QtiScalar) {
-                        continue;
-                    }
-
-                    if ($variable->getIdentifier() === 'SCORE_TOTAL') {
-                        $scoreTotal = $value->getValue();
-                    }
+        $launchData = $this->getLtiContextRepository()->findByDeliveryExecutionId($event->getDeliveryExecutionId());
+
```

Comment:
```suggestion
if (!$launchData)
{
   return; 
}
```
no LTI context - no AGS

Reply:
Yes good point, added.